homeassistant/components/hue/scene.py

```diff
@@ -0,0 +1,121 @@
+        raise NotImplementedError("Scene support is only available for V2 bridges")
+
+    if not bridge.allow_scenes:
+        return
+
+    # add entities for all scenes
+    @callback
+    def async_add_entity(event_type: EventType, resource: HueScene) -> None:
+        """Add entity from Hue resource."""
+        async_add_entities([HueSceneEntity(bridge, api.scenes, resource)])
+
+    # add all current items in controller
+    for item in api.scenes:
+        async_add_entity(EventType.RESOURCE_ADDED, item)
+
+    # register listener for new items only
+    config_entry.async_on_unload(
+        api.scenes.subscribe(async_add_entity, event_filter=EventType.RESOURCE_ADDED)
+    )
+
+
+class HueSceneEntity(HueBaseEntity, SceneEntity):
+    """Representation of a Scene entity from HUE Scenes."""
+
+    def __init__(
+        self,
+        bridge: HueBridge,
+        controller: ScenesController,
+        resource: HueScene,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(bridge, controller, resource)
+        self.resource = resource
+        self.controller = controller
+
+    @property
+    def name(self) -> str:
+        """Return default entity name."""
+        group = self.controller.get_group(self.resource.id)
+        return f"{group.metadata.name} - {self.resource.metadata.name}"
+
+    @property
+    def is_dynamic(self) -> bool:
+        """Return if this scene has a dynamic color palette."""
+        if self.resource.palette.color and len(self.resource.palette.color) > 1:
+            return True
+        if (
+            self.resource.palette.color_temperature
+            and len(self.resource.palette.color_temperature) > 1
+        ):
+            return True
+        return False
+
+    async def async_activate(self, **kwargs: Any) -> None:
+        """Activate Hue scene."""
+        transition = kwargs.get("transition")
+        if transition is not None:
+            # hue transition duration is in steps of 100 ms
+            transition = int(transition * 100)
+        dynamic = kwargs.get("dynamic", self.is_dynamic)
+        await self.bridge.async_request_call(
+            self.controller.recall,
+            self.resource.id,
+            dynamic=dynamic,
+            duration=transition,
+        )
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return the optional state attributes."""
+        group = self.controller.get_group(self.resource.id)
+        brightness = None
+        if palette := self.resource.palette:
+            if palette.dimming:
+                brightness = palette.dimming[0].brightness
+        if brightness is None:
+            # get brightness from actions
+            for action in self.resource.actions:
+                if action.action.dimming:
+                    brightness = action.action.dimming.brightness
+                    break
+        return {
+            "is_hue_scene": True,
```

Comment:
This is weird ?

Reply:
You mean "is_hue_scene" ?
I did that because in the previous implementation each grouped light had the same attribute "is_hue_group"
If it's not needed, I'll remove off course because it doesn't add a thing

Reply:
if I may, leaving this in allows easy templating based in that attribute, and also in dev-tools state, finding all Hue scenes, and disregarding all others.
as a matter of fact, I have been experimenting with that lately, and that works wonders. Please keep it?

Reply:
Lights have it because there are 2 types of lights provided by Hue: groups and non-groups. 

We should not have integrations provide their own unique ways of identifying the integration. Instead a generic way to identify the source of an entity could be added but not in this PR.
```suggestion
```

Reply:
I agree with @balloob if it is of any relevance to know to which integration an entity belongs to we could consider adding that in a more generic way